samples/excel/89-preview-apis/events-worksheet-activated.yaml

```diff
@@ -0,0 +1,156 @@
+order: 7
+id: excel-events-worksheet-activated
+name: Events - Worksheet activated
+description: Add an event handler on the worksheet activated and deactivated events
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |+
+        $("#setup").click(() => tryCatch(setup));
+        $("#activate-event-handler").click(() => tryCatch(onActivateEventHandler));
+        $("#deactivate-event-handler").click(() => tryCatch(onDeactivateEventHandler));
+        $("#delete-worksheet").click(() => tryCatch(deleteWorksheet));
+
+        async function onActivateEventHandler() {
+            await Excel.run(async (context) => {
+                let sheet = context.workbook.worksheets;
+                sheet.onActivated.add(onActivate);
+
+                await context.sync();
+
+                OfficeHelpers.UI.notify("Registered worksheet on activated event handler.");   
+            });
+        }
+
+        function onActivate(args) {
+            console.log("The activated worksheet Id : " + args.worksheetId);
+        }
+
+        async function onDeactivateEventHandler() {
+            await Excel.run(async (context) => {
+                let sheet = context.workbook.worksheets;
+                sheet.onDeactivated.add(onDeactivate);
+
+                await context.sync();
+
+                OfficeHelpers.UI.notify("Registered worksheet on activated event handler.");
+            });
+        }
+
+        function onDeactivate(args) {
+            console.log("The deactivated worksheet Id : " + args.worksheetId);
+        }
+
+        async function deleteWorksheet() {
+            await Excel.run(async (context) => {
+                let sheet = context.workbook.worksheets;
+                sheet.load("items/name");
+
+                await context.sync();
+
+                if (sheet.items.length > 1) {
```

Comment:
Another example of why "sheet" singular is confusing. The line `sheet.items.length` implies that a single worksheet has a collection of items (of some mysterious type). What's meant here is the collection of worksheets in the parent workbook.